lib/src/blaze/fetch_full_transaction.rs

```diff
@@ -404,6 +406,50 @@ impl TransactionContext {
                 .clone()
                 .try_into()
                 .unwrap_or(Memo::Future(memo_bytes));
+            if let Memo::Arbitrary(ref wallet_internal_data) = memo {
+                let mut uas_bytes = vec![0; 511];
+                wallet_internal_data
+                    .as_slice()
+                    .read(&mut uas_bytes)
+                    .unwrap();
+                let uas_string = String::from_utf8(uas_bytes);
+                if let Ok(uas_string) = uas_string {
+                    for ua_string in uas_string.split('!') {
```

Comment:
Use HRP! to delimit to save space:  https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki#bech32